toothpick-runtime/src/main/java/toothpick/Toothpick.java

```diff
@@ -121,7 +122,9 @@ public static void closeScope(Object name) {
    */
 
   public static void reset() {
-    MAP_KEY_TO_SCOPE.clear();
+    for (Object name : new ArrayList<>(MAP_KEY_TO_SCOPE.keySet())) {
+      closeScope(name);
+    }
```

Comment:
so scopes closed this way:

1. are properly closed.
2. and cannot be opened after closed either.